edit-post/hooks/autocompleters/index.js

```diff
@@ -0,0 +1,51 @@
+/**
+ * External dependencies.
+ */
+import { find } from 'lodash';
+
+/**
+ * WordPress dependencies.
+ */
+import { addFilter } from '@wordpress/hooks';
+import { select, dispatch } from '@wordpress/data';
+import { createBlock } from '@wordpress/blocks';
+
+// Export for unit test
+export function getBlockOptions( query ) {
+	const editor = select( 'core/editor' );
+	const isInitialQuery = ! query;
+
+	return isInitialQuery ?
+		// Before we have a query, offer frecent blocks as a sensible default.
+		editor.getFrecentInserterItems() :
+		editor.getInserterItems();
+}
+
+// Export for unit test
+export function getBlockCompletion( inserterItem ) {
+	const { name, initialAttributes } = inserterItem;
+	return {
+		action: 'replace',
+		value: createBlock( name, initialAttributes ),
+	};
+}
+
+// Export for unit test
+export function addReusableBlocksCompletion( completers ) {
+	const blocksCompleter = find( completers, c => 'blocks' === c.name );
+	if ( blocksCompleter ) {
+		blocksCompleter.options = getBlockOptions;
+		blocksCompleter.getOptionCompletion = getBlockCompletion;
+
+		// Fetch shared block data so it will be available to this completer later.
+		dispatch( 'core/editor' ).fetchSharedBlocks();
```

Comment:
I think it's okey as a quick temporary solution. However, it the long run we should avoid introducing this kind of changes and refactor `fetchSharedBlocks` to be hidden behind new `wp.data` API. It's a perfect fit to be implemented as `resolver`. See: https://github.com/WordPress/gutenberg/tree/master/data#registering-a-store.

Reply:
It might be hard to refactor shared blocks to use `wp.data` now that its implementation is pretty tied up with how blocks are stored in redux (see https://github.com/WordPress/gutenberg/pull/5228).

Reply:
@aduth or @youknowriad can you confirm that `wp.data` doesn't make sense in this context?

Reply:
There might be a misunderstanding here: `wp.data` is effectively an abstraction over Redux. I don't think the two are incompatible, and I'm inclined to agree that triggering a fetch here is not the best approach.